network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -360,6 +362,82 @@ private boolean isAnOpenSwitch(Edge edge) {
+                    return resource.getAttributes().getLeg2().getNode();
+                } else {
+                    return resource.getAttributes().getLeg3().getNode();
+                }
+            })
+            .collect(Collectors.toSet());
+
+        Set<Integer> generatorNodes = index.getStoreClient().getVoltageLevelGenerators(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> batteryNodes = index.getStoreClient().getVoltageLevelBatteries(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> loadNodes = index.getStoreClient().getVoltageLevelLoads(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> shuntCompensatorNodes = index.getStoreClient().getVoltageLevelShuntCompensators(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> staticVarCompensatorNodes = index.getStoreClient().getVoltageLevelStaticVarCompensators(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> danglingLineNodes = index.getStoreClient().getVoltageLevelDanglingLines(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> lccConverterStationNodes = index.getStoreClient().getVoltageLevelLccConverterStations(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> vscConverterStationNodes = index.getStoreClient().getVoltageLevelVscConverterStations(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
+        Set<Integer> groundNodes = index.getStoreClient().getVoltageLevelGrounds(index.getNetwork().getUuid(), index.getWorkingVariantNum(), voltageLevelResource.getId())
+            .stream().map(resource -> resource.getAttributes().getNode())
+            .collect(Collectors.toSet());
```

Comment:
all lines below seem to be simiar can it be refactored ?

Reply:
Done